llvm/lib/Transforms/Scalar/MemCpyOptimizer.cpp

```diff
@@ -1854,12 +1855,53 @@ bool MemCpyOptPass::processMemCpy(MemCpyInst *M, BasicBlock::iterator &BBI) {
   return false;
 }
 
+/// Memmove calls with overlapping src/dest buffers that come after a memset may
+/// be removed.
+bool MemCpyOptPass::mayBeMemMoveMemSetDependency(MemMoveInst *M) {
```

Comment:
I guess this should be "must"?

Reply:
I think it's implicit in the fact that the memmove comes after the memset?

Reply:
`may` in the function name is not strong enough, it should be something like `isMemMove...`